core/execution/future/special_orders.go

```diff
@@ -142,207 +139,38 @@ func (m *Market) repriceAllSpecialOrders(
 		}
 	}
 
-	// just checking if we need to take all lp of the book too
-	// normal lp updates would be fine without taking order from the
-	// book as no prices would be conlficting
 	needsPeggedUpdates := len(parked) > 0 || len(toSubmit) > 0
 
-	// first we save all the LP orders into the liquidity engine so that it can
-	// know the history during Update
-	m.liquidity.SaveLPOrders()
-	defer m.liquidity.ClearLPOrders()
-
-	// now we get the list of all LP orders, and get them out of the book
-	lpOrders := m.matching.GetAllLiquidityOrders()
-	m.removeLPOrdersFromBook(ctx, lpOrders)
-
-	// now no lp orders are in the book anymore,
-	// we can then just re-submit all pegged orders
-	// if we needed to re-submit pegged orders,
-	// let's do it now
-	if needsPeggedUpdates && len(toSubmit) > 0 {
-		updatedOrders, partiesPos := m.reSubmitPeggedOrders(ctx, toSubmit)
-		risks, _, _ := m.updateMargins(ctx, partiesPos)
-		if len(risks) > 0 {
-			transfers, distressed, _, err := m.collateral.MarginUpdate(
-				ctx, m.GetID(), risks)
-			if err == nil && len(transfers) > 0 {
-				evt := events.NewLedgerMovements(ctx, transfers)
-				m.broker.Send(evt)
-			}
-			for _, p := range distressed {
-				distressedParty := p.Party()
-				for _, o := range updatedOrders {
-					if o.Party == distressedParty && o.Status == types.OrderStatusActive {
-						// cancel only the pegged orders, the reset will get picked up during regular closeout flow if need be
-						_, err := m.cancelOrder(ctx, distressedParty, o.ID)
-						if err != nil {
-							m.log.Panic("Failed to cancel order",
-								logging.Error(err),
-								logging.String("OrderID", o.ID))
-						}
-					}
-				}
-			}
-		}
-	}
-
-	// now we have all the re-submitted pegged orders and the
-	// parked pegged orders from before
-	// we can call Update, which is going to give us the
-	// actual updates to be done on liquidity orders
-	newOrders, cancels := m.liquidity.Update(
-		ctx, minLpPrice, maxLpPrice, m.repriceLiquidityOrder)
-
-	return m.updateLPOrders(ctx, lpOrders, newOrders, cancels)
-}
-
-func (m *Market) enterAuctionSpecialOrders(
-	ctx context.Context,
-	updatedOrders []*types.Order,
-) {
-	// first remove all GFN orders from the peg list
-	ordersEvts := m.peggedOrders.EnterAuction(ctx)
-	m.broker.SendBatch(ordersEvts)
-
-	m.stopAllSpecialOrders(ctx, updatedOrders)
-}
-
-func (m *Market) stopAllSpecialOrders(
-	ctx context.Context,
-	updatedOrders []*types.Order,
-) {
-	// Park all pegged orders
-	updatedOrders = append(
-		updatedOrders,
-		m.parkAllPeggedOrders(ctx)...,
-	)
```

Comment:
some form of this function is still needed for pegged orders

Reply:
It's the `parkAllPeggedOrders` function. It's called elsewhere.